src/uct/base/uct_iface.c

```diff
@@ -1034,6 +1037,163 @@ int uct_iface_local_is_reachable(uct_iface_local_addr_ns_t *addr_ns,
     return 1;
 }
 
+struct route_info {
+    int if_index;
+    int family;
+
+    union {
+        struct in_addr  ipv4;
+        struct in6_addr ipv6;
+    } remote_addr;
+
+    int prefix_len;
+    int reachable;
+};
+
+static int netlink_get_route_info(int **if_idx, void **dst_in_addr,
```

Comment:
IMO `uct_iface.c` is not the place for it (in fact, it has nothing UCX specific).
Anyway, after moving, don't forget to prefix func names with module name.